pkg/kubectl/cmd/taint/utils.go

```diff
@@ -56,52 +66,51 @@ func parseTaints(spec []string) ([]corev1.Taint, []corev1.Taint, error) {
 			uniqueTaints[newTaint.Effect].Insert(newTaint.Key)
 
 			taints = append(taints, newTaint)
-		} else if strings.HasSuffix(taintSpec, "-") {
-			taintKey := taintSpec[:len(taintSpec)-1]
-			var effect corev1.TaintEffect
-			if strings.Index(taintKey, ":") != -1 {
-				parts := strings.Split(taintKey, ":")
-				taintKey = parts[0]
-				effect = corev1.TaintEffect(parts[1])
-			}
-
-			// If effect is specified, need to validate it.
-			if len(effect) > 0 {
-				err := validateTaintEffect(effect)
-				if err != nil {
-					return nil, nil, err
-				}
-			}
-			taintsToRemove = append(taintsToRemove, corev1.Taint{Key: taintKey, Effect: effect})
-		} else {
-			return nil, nil, fmt.Errorf("unknown taint spec: %v", taintSpec)
 		}
 	}
 	return taints, taintsToRemove, nil
 }
 
-// parseTaint parses a taint from a string. Taint must be of the format '<key>=<value>:<effect>'.
+// parseTaint parses a taint from a string, whose form must be either
+// '<key>=<value>:<effect>', '<key>:<effect>', or '<key>'.
```

Comment:
Why `<key>` is considered a valid taint here? It looks like it gets rejected later in `parseTaints` when adding a taint, but it is accepted when removing a taint. If this behavior is desired, it should be documented in the comments and the release note.

Reply:
I just read [your comment](https://github.com/kubernetes/kubernetes/pull/74159#issuecomment-466575717) above. It is fine, but please update comments and the release note to point it out.

Reply:
I updated the comment to `ParseTaints`, since that's the exported method.

I do want to point out that the exported behavior has not changed. Prior to this PR, the `ParseTaints` function parsed some forms (including  `<key>`) itself, but delegated parsing of other forms to `parseTaint`. This PR moves all parsing to `parseTaint`.

Given this, do you think updating the release note is necessary?